### PR TITLE
[7.x] Add allowedExceptions to queued jobs

### PR DESCRIPTION
**Pull request by themsaid**
This PR does 2 things:

1- Adds a UUID to job payload.
2- Adds a `maxExceptions` to the payload.

`maxExceptions` is set by defining a `$allowedExceptions` property in the jobs class.

Each time the job fails with an unhandled exception, the worker will check for the maximum exceptions allowed and fail the job if it exceeds this limit and ignore the attempts.

So if `$allowedExceptions = 2` and `$tries = 10`. The worker will retry the job 10 times if it was released without an exception, but only 2 times if it was released with an exception.

### CODE REVIEW

**Comment by antonkomarev** [comment]
What is a use case?

**Comment by taylorotwell** [comment]
@antonkomarev it's to solve this problem a bit differently and across all drivers: https://github.com/laravel/framework/pull/31537

**Comment by andershagbard** [comment]
Will the job be released instantly, or is it possible to set amount of seconds?

**Comment by themsaid** [comment]
@andershagbard you can use --delay or delaySeconds to control the delay.

File: tests/Queue/RedisQueueIntegrationTest.php [hunk 1]
```diff
@@ -226,22 +226,22 @@ public function testBlockingPopProperlyPopsJobOffOfRedis($driver)
...
-        $this->queue->later(-200, $jobs[0]);
-        $this->queue->later(-200, $jobs[1]);
-
-        $this->assertEquals($jobs[0], unserialize(json_decode($this->queue->pop()->getRawBody())->data->command));
-        $this->assertEquals($jobs[1], unserialize(json_decode($this->queue->pop()->getRawBody())->data->command));
-
-        $this->assertEquals(0, $this->redis[$driver]->connection()->llen('queues:default:notify'));
-        $this->assertEquals(0, $this->redis[$driver]->connection()->zcard('queues:default:delayed'));
-        $this->assertEquals(2, $this->redis[$driver]->connection()->zcard('queues:default:reserved'));
+//        $this->setQueue($driver, 'default', null, 60, 5);
```
Comment (themsaid) [inline tests/Queue/RedisQueueIntegrationTest.php]: This test seems to fail randomly. Sometimes it passes and other times it fails. I can't identify what's going on.

**Comment by pawlox** [comment]
Hm, any example how to achieve requeue functionality with that? I mean when a job is released and it's not a "real" attempt but throttling attempt I don't want to increment job's attemps.

**Pull request merged**